### PR TITLE
Avoid redirecting kcat error output, as it doesn't honour the JSON format

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

When running `kcat -b ... -L -J`, a JSON output is expected, but `kcat` prints in its stderr non-json error log messages.

As one of the steps expect a JSON, separating stderr and stdout seems legit.

## Type of change

- Refactor (refactoring code, removing useless files)

## Testing steps

Tested locally

## Checklist
* [x] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [ ] updated documentation wherever necessary
* [ ] new tests can be executed both locally and within docker container
* [ ] new tests have been included in scenario list (make update-scenarios)

### CODE REVIEW

**Review by juandspy — APPROVED** [review]

**Review by ikerreyes — COMMENTED** [review]
But if you redirect stderr to stdout, then it should also be fine, right?

File: features/steps/kafka_steps.py [hunk 1]
```diff
@@ -41,25 +41,30 @@ def retrieve_broker_metadata(context, hostname=None, port=None):
...
 
     stdout_file = filepath_from_context(context, "logs/insights-results-aggregator/", "_stdout")
+    stderr_file = filepath_from_context(context, "logs/insights-results-aggregator/", "_stderr")
 
     # interact with the process:
     # read data from stdout and stderr, until end-of-file is reached
     stdout, stderr = out.communicate()
 
-    assert stderr is None, "Error during check"
     assert stdout is not None, "No output from process"
```
Comment (ikerreyes) [inline features/steps/kafka_steps.py:54]: If the output is expected in stderr, can it be now that stdout is empty?

File: pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [tool.ruff]
+line-length = 100
```
Comment (ikerreyes) [inline pyproject.toml:2]: I think we should do this in a different MR

File: features/steps/kafka_steps.py (hunk 1 above)
Comment (joselsegura) [reply features/steps/kafka_steps.py:54]: No, as both outputs are redirected to a "pipe", both variables should be valid, even if they are empty.

Now the output won't be expected in stderr: this change makes that both output are separated and treated different. Previously, the stderr was redirected to the stdout, so all the messages were received in the same `stdout` variable, making `stderr` to contain a `None`

**Review by matysek — APPROVED** [review]

**Review by ikerreyes — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (joselsegura) [reply pyproject.toml]: Oops, updated by mistake. I'll include in a separated one to avoid some warning when running ruff
